hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/MergeOnReadSnapshotRelation.scala

```diff
@@ -111,7 +112,7 @@ abstract class BaseMergeOnReadSnapshotRelation(sqlContext: SQLContext,
     val optionalFilters = filters
     val readers = createBaseFileReaders(tableSchema, requiredSchema, requestedColumns, requiredFilters, optionalFilters)
 
-    new HoodieMergeOnReadRDD(
+    new HoodieMergeOnReadRDDV2(
```

Comment:
specifically stuff like this where checkpoint affects sth we use for snapshot query. is a side effect to try and avoid